core/domain/acl_decorators.py

```diff
@@ -872,6 +872,62 @@ def test_can_edit_story(self, topic_id, **kwargs):
     return test_can_edit_story
 
 
+def can_edit_skill(handler):
+    """Decorator to check whether the user can edit a skill, which can be
+    independent or belong to a topic.
+    """
+    def test_can_edit_skill(self, **kwargs):
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+
+        user_actions_info = user_services.UserActionsInfo(self.user_id)
+        if role_services.ACTION_EDIT_OWNED_TOPIC in user_actions_info.actions:
+            return handler(self, **kwargs)
+        else:
+            raise self.UnauthorizedUserException(
+                'You do not have credentials to edit this skill.')
+
+    test_can_edit_skill.__wrapped__ = True
+    return test_can_edit_skill
+
+
+def can_delete_skill(handler):
+    """Decorator to check whether the user can delete a skill.
+    """
+    def test_can_delete_skill(self, **kwargs):
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+
+        user_actions_info = user_services.UserActionsInfo(self.user_id)
+        if role_services.ACTION_EDIT_ANY_TOPIC in user_actions_info.actions:
```

Comment:
ACTION_DELETE_ANY_SKILL

In general avoid conflating actions implicitly. Use the role assignments in role_services.py to manage who can do what.

Reply:
Done.